lib/game/widget_builder.dart

```diff
@@ -0,0 +1,202 @@
+          : null,
+      onHorizontalDragUpdate: game is HorizontalDragDetector
+          ? (DragUpdateDetails d) =>
+              (game as HorizontalDragDetector).onHorizontalDragUpdate(d)
+          : null,
+      onHorizontalDragEnd: game is HorizontalDragDetector
+          ? (DragEndDetails d) =>
+              (game as HorizontalDragDetector).onHorizontalDragEnd(d)
+          : null,
+      onHorizontalDragCancel: game is HorizontalDragDetector
+          ? () => (game as HorizontalDragDetector).onHorizontalDragCancel()
+          : null,
+
+      // Force presses
+      onForcePressStart: game is ForcePressDetector
+          ? (ForcePressDetails d) =>
+              (game as ForcePressDetector).onForcePressStart(d)
+          : null,
+      onForcePressPeak: game is ForcePressDetector
+          ? (ForcePressDetails d) =>
+              (game as ForcePressDetector).onForcePressPeak(d)
+          : null,
+      onForcePressUpdate: game is ForcePressDetector
+          ? (ForcePressDetails d) =>
+              (game as ForcePressDetector).onForcePressUpdate(d)
+          : null,
+      onForcePressEnd: game is ForcePressDetector
+          ? (ForcePressDetails d) =>
+              (game as ForcePressDetector).onForcePressEnd(d)
+          : null,
+
+      // Pan
+      onPanDown: game is PanDetector
+          ? (DragDownDetails d) => (game as PanDetector).onPanDown(d)
+          : null,
+      onPanStart: game is PanDetector
+          ? (DragStartDetails d) => (game as PanDetector).onPanStart(d)
+          : null,
+      onPanUpdate: game is PanDetector
+          ? (DragUpdateDetails d) => (game as PanDetector).onPanUpdate(d)
+          : null,
+      onPanEnd: game is PanDetector
+          ? (DragEndDetails d) => (game as PanDetector).onPanEnd(d)
+          : null,
+      onPanCancel: game is PanDetector
+          ? () => (game as PanDetector).onPanCancel()
+          : null,
+
+      // Scales
+      onScaleStart: game is ScaleDetector
+          ? (ScaleStartDetails d) => (game as ScaleDetector).onScaleStart(d)
+          : null,
+      onScaleUpdate: game is ScaleDetector
+          ? (ScaleUpdateDetails d) => (game as ScaleDetector).onScaleUpdate(d)
+          : null,
+      onScaleEnd: game is ScaleDetector
+          ? (ScaleEndDetails d) => (game as ScaleDetector).onScaleEnd(d)
+          : null,
+
+      child: Container(
+          color: game.backgroundColor(),
+          child: Directionality(
+              textDirection: TextDirection.ltr,
+              child: EmbeddedGameWidget(game))),
+    );
+  }
+}
+
+class OverlayGameWidget extends StatefulWidget {
+  final Widget child;
+  final HasWidgetsOverlay game;
+
+  OverlayGameWidget({this.child, this.game});
+
+  @override
+  State<StatefulWidget> createState() => _OverlayGameWidgetState(game);
+}
+
+class _OverlayGameWidgetState extends State<OverlayGameWidget> {
+  final Map<String, Widget> _overlays = {};
+
+  _OverlayGameWidgetState(HasWidgetsOverlay game) {
```

Comment:
Don't pass stuff to state's contractor since the state lives longer than the widget (therefore `createState` method doesn't run in every render).

You can access the game with `widget.game`